src/test/scala/is/hail/utils/IntervalSuite.scala

```diff
@@ -0,0 +1,211 @@
+package is.hail.utils
+
+import is.hail.annotations.ExtendedOrdering
+import is.hail.expr.types.TInt32
+import org.testng.annotations.Test
+
+class IntervalSuite {
+
+  val pord: ExtendedOrdering = TInt32().ordering
+
+  // set of intervals chosen from 5 endpoints spans the space of relations
+  // that two non-empty intervals can have with each other.
+  val points: IndexedSeq[Int] = 1 to 5
+
+  val test_intervals: IndexedSeq[SetInterval] =
+    for {
+      s <- points
+      e <- points
+      is <- Array(true, false)
+      ie <- Array(true, false)
+    } yield SetInterval(s, e, is, ie)
+
+  val test_itrees: IndexedSeq[SetIntervalTree] =
+    for {
+      i1 <- test_intervals
+      i2 <- test_intervals
+    } yield SetIntervalTree(Array(i1, i2).zipWithIndex)
+
+  def shouldOrShouldnt(bool: Boolean): String = if (bool) "should" else "shouldn't"
+
+  @Test def intervalEmpty() {
+    for (set_interval <- test_intervals) {
+      val interval = set_interval.interval
+      assert(interval.definitelyEmpty(pord) iff set_interval.definitelyEmpty(),
+        s"$interval ${ shouldOrShouldnt(set_interval.definitelyEmpty()) } be empty")
```

Comment:
I'm skeptical about these kinds of messages unless they contain specific information that can be used to debug the test, for example: "interval [3, 3), definitelyEmpty = false, expected true".  Part of the point of the Java/Scala test suite magic assert functions (or the different assert functions like assertEqual in Python unittest) is to generate these kinds of messages automatically.  What's more, it should only take a few seconds to get a failed test up in the debugger and see these values, so I'm not convinced of the value in making the code more complicated instead of focusing on us getting batter at using the tooling and keeping the code simpler.  I'm starting some dev posts to discuss testing standards, let's take this up there.  I'm not sure this will be settled now, but I thought I'd share my thoughts.

Reply:
Some of these won't throw sensical messages, even with the TestNG Assert methods, so I think having a message can still be valuable. I found them pretty helpful when I was debugging overlaps, because the set of intervals that it fails on basically tells you what's happening, but I'll at least go through and change all of them to be better asserts and prune the ones that don't give any more information than the default.